Add a handbook search command

Birdulon opened the pull request:
## Description
Adds a `/handbook` command for quick searching of ids.
![image](https://user-images.githubusercontent.com/5397662/167443259-0642ebf8-1c3a-4638-be39-5923ed3ff823.png)

Searching is case-insensitive and strips all non-letter characters.

No idea how this will work on other languages. (also I haven't done the language file stuff yet)

Should probably be mostly offloaded into some Utils wrapper to better allow integration with other commands (i.e. let other commands take string parameters instead of itemIDs and succeed if there's one match, or print all matches and ask for the command again with a more specific query)

## Issues fixed by this PR

<!--- Put the links of issues that may be fixed by this PR here (if any). -->
## Type of changes

<!--- Put an `x` in all the boxes that apply your changes. -->

- [ ] Bug fix
- [x] New feature 
- [ ] Enhancement
- [ ] Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md)
- [x] I am responsible for any copyright issues with my code if it occurs in the future.

Birdulon committed:
Add a handbook search command

KingRainbow44 commented:
coming back to this, it is planned to implement string query support for ID-based commands, its just not a high priority rn

Birdulon commented:
> coming back to this, it is planned to implement string query support for ID-based commands, its just not a high priority rn

I'm leaving that for when something like #751 goes through and we get something that can evaluate parameters so that we can have exchanges like:

```
>spawn hypostasis
[WARNING] Multiple results for "hypostatis" - choose from the list below or press N to cancel:
// Monsters
[1] 20040201 : Anemo Hypostasis - Beth
[2] 20040601 : Pyro Hypostasis: Ayin
[3] 20011701 : Hydro Hypostasis Summon: Split (S)
[4] 20011601 : Hydro Hypostasis Summon: Split (L)
[5] 20040501 : Cryo Hypostasis - Daleth
[6] 20011901 : Hydro Hypostasis Summon: Heal
[7] 20011801 : Hydro Hypostasis Summon: Self-Destruct
[8] 20040401 : Hydro Hypostasis
[9] 20040202 : (Event) Anemo Hypostasis - Beth
[10] 20040102 : (Event) Electro Hypostasis - Aleph
[11] 20040301 : Geo Hypostasis: Gimel
[12] 20040101 : Electro Hypostasis - Aleph
[13] 20040302 : (Event) Geo Hypostasis - Gimel
[WARNING] Additional search results which may not be useful for this argument:
// Items
[14] 210057 : Celebration: Hypostasis
>12
[WARNING] Target not specified! Specify a target or press N to cancel:
// Top 10 online players:
[1] @10002 Birdulon (127.0.0.1)
[2] @10003 Birdulon2 (127.0.0.1)
[3] @10004 Birdulon3 (127.0.0.1)
[4] @10005 Birdulon4 (127.0.0.1)
>@10007
[WARNING] UID @10007 is not currently online, would you like to schedule this event for when they log in next? N/y:
>y
[INFO] Scheduled "/spawn @10007 20040101" for when @10007 next logs in.
```

(ok that last bit might be a bit too far)

link-discord commented:
Would be most useful for the team command ngl

Would love to do something like

/team set 1 Ayaka

Then I wouldn't need to remember ids

KingRainbow44 commented:
Closing due to the current plans to implement direct name -> ID support in commands which use IDs.
